src/note.rs

```diff
@@ -139,6 +166,10 @@ pub struct Note {
     rho: Rho,
```

Comment:
str4d ([Link](https://github.com/zcash/orchard/pull/471#discussion_r2543612212)): Non-blocking: I think I'd prefer this to be
```suggestion
    rho: Option<Rho>,
```
and then remove `Rho::zero`, rather than opening up the possibility that `rho` accidentally gets used as zero in some codepath where `Note::update_rho_for_issuance_note` accidentally hasn't been called. We can then replace internal `self.rho` usage with `self.rho.expect("must call Note::update_rho_for_issuance_note first")` (panics are fine here as this is a programming error that should always be fixed if it occurs).